### ISSUE TITLE
Why can we not load images from VOC retrain (SSD)?

### ISSUE THREAD

**Issue by conner-starsky** [chainer/chainercv#416]
Why can we not use the weights learned in the VOC object detection? It seems like this would allow for much better accuracy than using weights that start from ImageNet. Would love to hear if this is a correct line of thought and what other people think.

**Comment by Hakuyume** [comment]
Let me confirm.
You want to create a SSD model that has different number of classes from VOC (more/less than 20) and initialize its weights by VOC trained model as much as possible. Is this right?

I think this feature is helpful and not difficult to implement. However, I'm wondering what is the best API.

Here are some candidates.
1. Do not change the arguments. If the `n_fg_class` is different from the number of class of `pretrained_model`, the initialization of final layers is skipped (initialized by random).
2. Add a new pretrained_model option, `voc0712_without_conf`.
3. Add a boolean argument. If it is `True`, the initialization of final layers is skipped. (`finetune_mode`?)

@conner-starsky @yuyu2172 How do you think it?

**Comment by yuyu2172** [comment]
Actually, I am working on this.
I think that the cleanest API choice is 1.

However, it gets complicated to load a pretrained weight in the current Chainer.
I am planning to add support for this feature after this PR is merged to the main Chainer.
https://github.com/chainer/chainer/pull/3176

**Comment by Hakuyume** [comment]
> I think that the cleanest API choice is 1.

I agree that 1 is most simple. One concern is that users cannot ignore the final layers when the number of classes is 20 but the classes are different from those of VOC. Perhaps, we can change the condition from `the n_fg_class is different from the number of class of pretrained_model`  to `when n_fg_class is specified`?

> However, it gets complicated to load a pretrained weight in the current Chainer.
I am planning to add support for this feature after this PR is merged to the main Chainer.
chainer/chainer#3176

Nice job.

**Comment by conner-starsky** [comment]
I like 1 as well.

**Comment by conner-starsky** [comment]
Currently, I am having a problem with the SSD demo, where the training script just freezes and doesn't seem to do anything (even with a Titan X selected).

**Comment by Hakuyume** [comment]
> Currently, I am having a problem with the SSD demo, where the training script just freezes and doesn't seem to do anything (even with a Titan X selected).

Please try `MultiprocessIterator` -> `SerialIterator`.
https://github.com/chainer/chainercv/blob/master/examples/ssd/train.py#L154

**Comment by conner-starsky** [comment]
It works now, thank you. Still seems a little bit slow at `0.58801 iters/sec`

**Comment by Hakuyume** [comment]
> It works now, thank you. Still seems a little bit slow at 0.58801 iters/sec

That speed is normal. Full-training of SSD takes 2 or 3 days.

**Comment by conner-starsky** [comment]
I guess this highlights while a partial retrain is needed :wink:

**Comment by yuyu2172** [comment]
>  One concern is that users cannot ignore the final layers when the number of classes is 20 but the classes are different from those of VOC.

I see. Maybe adding another option is less error prone.

**Comment by sanyuwen** [comment]
I want install cupy on windows.
is it ok? is it supported on windows 10?
I ran pip install cupy.
it give me an error.
error: Microsoft Visual C++ 14.0 is required.

so the cupy dose not support visual studio 2017,right?

**Comment by yuyu2172** [comment]
Some relevant PRs are merged to Chainer.
https://github.com/chainer/chainer/pull/4682

**Comment by yuyu2172** [comment]
We added a tutorial that discusses hwo to solve this problem.
